rfd/0232-desktop-access-protobuf-protocol.md

```diff
@@ -0,0 +1,596 @@
+    string username = 1;
+    ClientScreenSpec screen_spec = 2;
+    // Future initialization fields
+    // or capability advertisements here
+}
+
+message ServerHello {
+    ConnectionActivated activation_data = 1;
+    // Future initialization fields
+    // or capability advertisements here
+}
+```
+
+These messages will be exchanged at the start of the connection.
+
+```
++--------+                                +--------+
+| client |                                | server |
++--------+                                +--------+
+     |           Client Hello                 |
+     |--------------------------------------->|
+     |           Server Hello                 |
+     |<---------------------------------------|
+     |             PNG frame                  |
+     |<---------------------------------------|
+     |             Mouse Move                 |
+     |--------------------------------------->|
+```
+
+
+### Notable Message Updates
+
+#### 2 - PNG frame and 27 - PNG Frame 2 
+Messages 2 and 27, (PNG Frame and PNG Frame 2) will be consolidated into a
+single message. The optimization brought on by the PNG 2 message is obsoleted
+under protobufs.
+
+```protobuf
+message Rectangle {
+    uint32 left = 1;
+    uint32 top = 2;
+    uint32 right = 3;
+    uint32 bottom = 4;
+}
+
+message PNG frame {
+    Rectangle coordinates = 1;
+    bytes data = 2;
+}
+```
+
+#### Message 10 - MFA 
+The MFA message will no longer contain json. Instead, it will compose the
+existing `MFAAuthenticationChallenge` and `MFAAuthenticateResponse` messages.
+
+## Backwards Compatibility with Screen Recordings
+RFD 48 defines a protocol buffer message `DesktopRecordingEvent` that captures
+a subset of TDP messages required for session playback.  We can simply add a
+new field, `ProtoMessage` to this message.
+
+```protobuf
+message DesktopRecordingEvent {
+    // Metadata is a common event metadata
+    Metadata Metadata = 1;
+
+    // Message is the encoded TDP message. It is not marshaled to JSON format
+    bytes Message = 2;
+
+    // A TDPFrame message which contains PNG, Screen Spec,
+    // ClipboardData, Mouse Move, and mouse button events
+    bytes ProtoMessage = 4;
+
+    // DelayMilliseconds is the delay in milliseconds from the start of the session
+    int64 DelayMilliseconds = 3;
+}
+```
+
+The existing session player will need a small update to handle desktop recording
+messages containing either a TDP message or TDPB message.
+
+### Appendix A: Example Message Definitions
+```protobuf
+enum MessageType {
+  MESSAGE_UNKNOWN = 0;
+  MESSAGE_CLIENT_HELLO = 1;
+  MESSAGE_SERVER_HELLO = 2;
+  MESSAGE_RECTANGLE = 3;
+  MESSAGE_PNG_FRAME = 4;
+  MESSAGE_FASTPATH_PDU = 5;
+  MESSAGE_RDP_RESPONSE_PDU = 6;
+  MESSAGE_CONNECTION_ACTIVATED = 7;
+  MESSAGE_SYNC_KEYS = 8;
+  MESSAGE_MOUSE_MOVE = 9;
+  MESSAGE_MOUSE_BUTTON = 10;
+  MESSAGE_KEYBOARD_BUTTON = 11;
+  MESSAGE_CLIENT_SCREEN_SPEC = 12;
+  MESSAGE_CLIENT_USERNAME = 13;
```

Comment:
Similarly, I wonder if the client username and screen spec could be moved into the client hello?

Edit: looks like you've already moved the username into the client hello down below so maybe we just need a section under `Notable Message Updates` which states that the client username message doesn't exist in TDPB.